projects/js-packages/critical-css-gen/src/browser-interface-iframe.ts

```diff
@@ -140,7 +149,18 @@ export class BrowserInterfaceIframe extends BrowserInterface {
 			return;
 		}
 
-		const fullUrl = this.addGetParameters( rawUrl );
+		// Make sure URL is valid.
+		let url;
+		try {
+			url = new URL( rawUrl );
+		} catch ( err ) {
+			return new Promise( ( resolve, reject ) => {
+				this.trackUrlError( rawUrl, err );
+				reject( new InvalidURLError( { url: rawUrl } ) );
+			} );
```

Comment:
Nit: We could use the below instead here, since loadPage is async, it's the same as rejecting the promise, but minimises callback hell
```suggestion
                this.trackUrlError( rawUrl, err );
                throw new InvalidURLError( { url: rawUrl } );
```

Reply:
I've updated Liam's suggestion to fix the whitespace (spaces to tabs).

Reply:
@LiamSarsfield I had no idea this is what's making me not like JS. WOW - nice read. Thank you!